src/potential_file_reader.cpp

```diff
@@ -155,9 +157,21 @@ TextFileReader * PotentialFileReader::open_potential(const std::string& path) {
       utils::logmesg(lmp, fmt::format("Reading potential file {} with DATE: {}\n", filename, date));
     }
 
-    if (!units.empty() && (units != unit_style)) {
-      lmp->error->one(FLERR, fmt::format("Potential file {} requires {} units "
-                                         "but {} units are in use",filename, units, unit_style));
+    if (units.empty()) {
+      unit_convert == utils::NOCONVERT;
+    } else {
+      if (units == unit_style) {
+        unit_convert = utils::NOCONVERT;
+      } else {
+        if ((units == "metal") && (unit_style == "real") && (unit_convert | utils::METAL2REAL)) {
```

Comment:
can we put ENG in these keywords, e.g. ENG_M2R or ENG_METAL2REAL

Reply:
those keywords are designed to be applied for conversion of other properties as well. The idea was that when conversion of multiple properties is requested (via bitwise OR), say length *and* energy, then only the intersections of conversions that is available for all properties will be returned as viable conversions. for that the same bits would have to be used for all properties. Not sure if it ever comes to that, but I didn't want to limit it needlessly in the initial design.